core/controllers/topic_viewer.py

```diff
@@ -71,12 +72,32 @@ def get(self, topic_name):
         uncategorized_skill_ids = topic.get_all_uncategorized_skill_ids()
         subtopics = topic.get_all_subtopics()
 
+        assigned_skill_ids = topic.get_all_skill_ids()
+        skill_descriptions = skill_services.get_skill_descriptions_by_ids(
+            topic.id, assigned_skill_ids)
+
+        if self.user_id:
+            degrees_of_mastery = skill_services.get_multi_user_skill_mastery(
+                self.user_id, assigned_skill_ids)
+            sorted_skill_ids = sorted(
+                degrees_of_mastery,
+                key=lambda skill_id: degrees_of_mastery[skill_id],
+                reverse=True)
+        else:
+            degrees_of_mastery = {}
+            for skill_id in assigned_skill_ids:
+                degrees_of_mastery[skill_id] = None
+            sorted_skill_ids = assigned_skill_ids
+
         self.values.update({
             'topic_id': topic.id,
             'topic_name': topic.name,
             'canonical_story_dicts': canonical_story_dicts,
             'additional_story_dicts': additional_story_dicts,
             'uncategorized_skill_ids': uncategorized_skill_ids,
-            'subtopics': subtopics
+            'subtopics': subtopics,
+            'sorted_skill_ids': sorted_skill_ids,
```

Comment:
so since we have uncategorized_skill_ids in this json too, its a little confusing to understand what 'sorted_skill_ids' are ..
Also, why do we need to pass sorted_skill_ids, is it just so that we can show degrees of mastery in descending order? Can we not just pass the mastery information and do the sorting work in the frontend?

Reply:
You can use orderBy in angularJS to do that - https://docs.angularjs.org/api/ng/filter/orderBy

Reply:
uncategorized_skill_ids doesn't contain all skill ids assigned to a topic. If a skill is assigned to a subtopic, it becomes a categorized skill instead.

Reply:
Done, but orderBy can only be used on a list, so I used another way to sort in the frontend.

Reply:
As discussed, you are anyway converting the object to a list so you can use orderBy in the frontend

Reply:
Done.